mind/app/solver.py

```diff
@@ -87,35 +98,24 @@ def find_random_path(cities_list, starting_city, time_left):
         curr_city = cities_list[next_city]
     #TODO
     # spróbować to zwracać jako tuple return time_left, sum, path
-    return [time_left, sum, path]
+    return time_left, total, path
 
 
-def find_best_of_random_paths(df_cities: pd.DataFrame, df_paths: pd.DataFrame, df_time: pd.DataFrame, n=50):
+def find_best_of_random_paths(d: dict, working_time: int, n=50): # -> Output:
     """
     Returns list [time_left, sum, path] for the best of paths found in random walk.
+    :param d: dictionary {name : {neighbour1 : travel_time1, neighbour2 : travel_time2}}
+    :param working_time:
     :param n: number of trials for each vertex in random walk
     """
 
-    d = convert_to_dict(df_cities, df_paths)
-
-    # define empty dictionary containing {city_name : city_object}
-    cities_list = {}
-
-    for k in d.keys():
-        vec = df_cities.loc[df_cities['ID'] == k].values[0]
-        c = City(k, vec[1], vec[2], vec[3])
-        c.getNeighbours(d)
-        cities_list[k] = c
-
-    time = df_time['time'].values[0]
-
     best_paths = []
     for starting_city in list(d.keys()):
         lst = []
         for i in range(n):
-            lst.append(find_random_path(cities_list, starting_city, time))
+            lst.append(find_random_path(d, starting_city, working_time))
```

Comment:
Jeśli jest to duża pętla to zdefiniowanie przed nią 
```
add = lst.append
```
powinno poprawić performance.